x/oracle/keeper/querier.go

```diff
@@ -135,13 +135,13 @@ func queryTask(ctx sdk.Context, path []string, req abci.RequestQuery, k Keeper,
 	if err != nil {
 		return nil, sdkerrors.Wrap(sdkerrors.ErrJSONUnmarshal, err.Error())
 	}
-	task, err := k.GetTask(ctx, params.Contract, params.Function)
+	task, err := k.GetTask(ctx, types.NewTaskID(params.Contract, params.Function))
 	if err != nil {
 		return nil, err
 	}
-	res, err = codec.MarshalJSONIndent(legacyQuerierCdc, task)
-	if err != nil {
-		return nil, err
+	res, err = nil, fmt.Errorf("failed to cast to concrete task")
+	if smartContractTask, ok := task.(*types.Task); ok {
+		res, err = codec.MarshalJSONIndent(legacyQuerierCdc, *smartContractTask)
```

Comment:
Wouldn't it be simpler and more readable to return directly?

Reply:
Yeah, that looks good to me. will have it changed on next commit.